### PR TITLE
fix(serverless-offline-sqs) hostname resolved to 127.0.0.1

### PR DESCRIPTION
**Pull request by jurienhamaker**
I came across an old issue: #33 
The same issue happened to me.

It was resolved however with the wrong answer. In his case (like mine) he ran serverless inside a docker too. Which then connects to the elasticMQ docker through the docker network.

The `client.getQueueUrl` returns `http://localhost:9324` because it doesn't know it's own ip. Therefore it will try to listen to messages on `https://localhost:9324/queue/<name>` for new messages and throw the  `Error: connect ECONNREFUSED 127.0.0.1:9324`

In this fix I simply replace the hostname of the returned queueUrl with the endpoint configured in the config. This should resolve most issues when it comes to offline handling of messages.

A better solution is welcome (:

### CODE REVIEW

**Comment by ShaunParsons** [comment]
👍 Just opened an issue about this before seeing this PR: https://github.com/CoorpAcademy/serverless-plugins/issues/115

File: packages/serverless-offline-sqs/src/index.js [hunk 1]
```diff
@@ -208,7 +208,10 @@ class ServerlessOfflineSQS {
       await fromCallback(cb => client.createQueue(params, cb));
     }
 
-    const {QueueUrl} = await fromCallback(cb => client.getQueueUrl({QueueName}, cb));
+    let {QueueUrl} = await fromCallback(cb => client.getQueueUrl({QueueName}, cb));
+
+    const endpointWithoutHost = QueueUrl.replace(/^[a-z]{4,5}:\/{2}[a-z]{1,}:\d{1,4}.(.*)/, '$1');
+    QueueUrl = `${this.getConfig().endpoint}/${endpointWithoutHost}`;
```
Comment (ShaunParsons) [inline packages/serverless-offline-sqs/src/index.js]: Perhaps we could remove the request to AWS completely?

```js
const QueueUrl = `${this.getConfig().endpoint}/queue/${QueueName}`;
```

Comment (jurienhamaker) [reply packages/serverless-offline-sqs/src/index.js]: I was about to reply this to the main pr.

The reason behind this is that it does not have to necessarily be the same endpoint for each message broker. 

AWS SQS does the following:
`<host>/<id>/<queue-name>`

This does not apply to elasticMq, which does:
`<host>/queue/<queue-name`

Therefore the request is necessary, but the hostname will always be the same. So we do the request to fetch the path of the endpoint. And then replace it with the provided hostname/endpoint.

Comment (ShaunParsons) [reply packages/serverless-offline-sqs/src/index.js]: Good point. Cheers for the explanation :)

**Comment by jefer590** [comment]
Hello 👋 sorry to bother everyone but is there an update regarding merging this and publish it?

**Comment by ShaunParsons** [comment]
@AdrieanKhisbe any chance we can get this merged?

**Comment by devpascoe** [comment]
Can this be merged now?

**Comment by devpascoe** [comment]
Merge? 😘

**Comment by justinlazaro-iselect** [comment]
can we merged it now? thanks

**Comment by justinlazaro-iselect** [comment]
any update to merge this?

**Comment by natac13** [comment]
Is there a work-around for this??

**Comment by godu** [comment]
I prefer use `new URL` instead of `Regex`.
I publish it in `serverless-offline-sqs@6.0.0`.
Thank you for your analysis :)

**Pull request closed without merging**